src/angular-dayparts.js

```diff
@@ -26,6 +26,114 @@ angular.module('angular-dayparts', [])
+            ];
+
+            $scope.selectedPresetItem = $scope.presetItems[0].value;
+
+            /*
+            * Get week parts
+            */
+            var week = getWeekParts(0, 23);
+            var weekend = getWeekParts(0, 23, [{name: 'Sunday', position: 1}, {name: 'Saturday', position: 7}]);
+            var weekdays = getWeekParts(0, 23, angular.copy($scope.days.slice(1, -1)));
+            var businessHours = getWeekParts(9, 16, angular.copy($scope.days.slice(1, -1)));
+            var eveningHours = getWeekParts(18, 23);
+
+            function getWeekParts (start, end, days = angular.copy($scope.days)) {
+              var getHours = function (start, end) {
+                var hours = [];
+                for (var i = start; i <= end; i++) {
+                  hours.push(i);
+                }
+                return hours;
+              }
+              var hours = getHours(start, end);
+              var dayParts = [];
+              days.forEach(function (day) {
+                hours.forEach(function (hour) {
+                  dayParts.push(day.name + '-' + hour)
+                })
+              })
+              return dayParts;
+            }
+
+            /*
+            * Make grid selections based on selected preset
+            */
+            function setPreset (weekPart) {
+              $element.find('td').each(function(i, el){
+                if (_.includes(weekPart, $(el).data('time'))) {
+                  $(el).addClass(klass);
+                }
+              });
+              $scope.options.onChange(weekPart);
+            }
+
+            $scope.selectedPresetItemChanged = function () {
+              if ($scope.selectedPresetItem === 'week') {
+                $timeout(function () {
+                  deselect();
+                  setPreset(week);
+                }, 100);
+              }
+              else if ($scope.selectedPresetItem === 'weekend') {
+                $timeout(function () {
+                  deselect();
+                  setPreset(weekend);
+                }, 100);
+              }
+              else if ($scope.selectedPresetItem === 'weekdays') {
+                $timeout(function () {
+                  deselect();
+                  setPreset(weekdays);
+                }, 100);
+              }
+              else if ($scope.selectedPresetItem === 'businessHours') {
+                $timeout(function () {
+                  deselect();
+                  setPreset(businessHours);
+                }, 100);
+              }
+              else if ($scope.selectedPresetItem === 'eveningHours') {
+                $timeout(function () {
+                  deselect();
+                  setPreset(eveningHours);
+                }, 100);
+              }
+            }
+
+            /*
+            * Make preset selection based on grid seletions
+            */
+            function changeSelectedPresetItem (item) {
```

Comment:
I removed the `changeSelectedPresetItem()` function entirely and now setting the the `selectedPresetItem` directly. 🔥